module/epochs/qc_client.go

```diff
@@ -0,0 +1,174 @@
+	signer            sdkcrypto.Signer // signer used to sign vote transaction
+
+	account *sdk.Account   // account belonging to the collection node
+	client  *client.Client // flow-go-sdk client to access node
+}
+
+// NewQCContractClient returns a new client to the Quorum Certificate contract
+func NewQCContractClient(nodeID flow.Identifier, accountAddress string,
+	accountKeyIndex uint, accessAddress, qcContractAddress string, signer sdkcrypto.Signer) (*QCContractClient, error) {
+
+	address := sdk.HexToAddress(accountAddress)
+
+	// create a new instance of flow-go-sdk client
+	flowClient, err := client.New(accessAddress, grpc.WithInsecure())
+	if err != nil {
+		return nil, fmt.Errorf("could not create flow client: %w", err)
+	}
+
+	// get account for given address
+	account, err := flowClient.GetAccount(context.Background(), address)
+	if err != nil {
+		return nil, fmt.Errorf("could not get account: %w", err)
+	}
+
+	// check if account key index within range of keys
+	if len(account.Keys) <= int(accountKeyIndex) {
+		return nil, fmt.Errorf("given account key index is bigger than the number of keys for this account")
+	}
+
+	return &QCContractClient{
+		accessAddress:     accessAddress,
+		qcContractAddress: qcContractAddress,
+		client:            flowClient,
+		account:           account,
+		accountKeyIndex:   accountKeyIndex,
+		signer:            signer,
+		nodeID:            nodeID,
+	}, nil
+}
+
+// SubmitVote submits the given vote to the cluster QC aggregator smart
+// contract. This function returns only once the transaction has been
+// processed by the network. An error is returned if the transaction has
+// failed and should be re-submitted.
+func (c *QCContractClient) SubmitVote(ctx context.Context, vote *model.Vote) error {
+
+	// get latest sealed block to execute transaction
+	latestBlock, err := c.client.GetLatestBlock(ctx, true)
+	if err != nil {
+		return fmt.Errorf("could not get latest block from node: %w", err)
+	}
+
+	// attach submit vote transaction template and build transaction
+	tx := sdk.NewTransaction().
+		SetScript(templates.GenerateSubmitVoteScript(c.getEnvironment())).
+		SetGasLimit(1000).
+		SetReferenceBlockID(latestBlock.ID).
+		SetPayer(c.account.Address)
+
+	// add signature data to the transaction and submit to node
+	err = tx.AddArgument(cadence.NewString(hex.EncodeToString(vote.SigData)))
+	if err != nil {
+		return fmt.Errorf("could not add raw vote data to transaction: %w", err)
+	}
+
+	// sign payload using account signer
+	err = tx.SignPayload(c.account.Address, int(c.accountKeyIndex), c.signer)
+	if err != nil {
+		return fmt.Errorf("could not sign transaction: %w", err)
+	}
+
+	// submit signed transaction to node
+	txID, err := c.submitTx(tx)
+	if err != nil {
+		return fmt.Errorf("failed to submit transaction: %w", err)
+	}
+
+	// wait for transaction to be sealed
+	result := &sdk.TransactionResult{Status: sdk.TransactionStatusUnknown}
+	for result.Status != sdk.TransactionStatusSealed {
```

Comment:
Either in this component or in `QCVoter` we should probably add a timeout on the context as a sanity check. Access nodes should eventually return an `Expired` status, but if they don't we could be stuck in this loop forever.